src/tapis-hooks/src/authenticator/useLogin.ts

```diff
@@ -0,0 +1,74 @@
+import { useContext } from 'react';
+import { useMutation } from 'react-query';
+import { Authenticator } from '@tapis/tapis-typescript';
+import { queryHelper } from '../utils';
+import TapisContext from '../context';
+
+interface LoginHelperParams {
+  username: string,
+  password: string
+}
+
+type LoginParams = {
+  onSuccess?: (data: Authenticator.RespCreateToken) => any,
+  onError?: (error) => any
+} & LoginHelperParams
+
+const useLogin = () => {
+  const tapisContext = useContext(TapisContext);
+
+  // On successful login, save the token to the TapisContext state
+  const onSuccess = (response: Authenticator.RespCreateToken) => {
+    tapisContext.setAccessToken(response.result.access_token);
+  }
+ 
+  // This helper takes the username and password and assembles an API call
+  const loginHelper = ({ username, password }: LoginHelperParams): Promise<Authenticator.RespCreateToken>  => {
```

Comment:
Should we extract these helpers to their own module? As we write more of these I think there will be value in having a collection of practical tapis-ui calls for all the operations.